tests/nvidiagpu/gpu_suite_test.go

```diff
@@ -27,4 +34,29 @@ func TestGPUDeploy(t *testing.T) {
 var _ = JustAfterEach(func() {
 	reporter.ReportIfFailed(
 		CurrentSpecReport(), currentFile, tsparams.ReporterNamespacesToDump, tsparams.ReporterCRDsToDump, clients.SetScheme)
+
+	dumpDir := inittools.GeneralConfig.GetDumpFailedTestReportLocation(currentFile)
+	if dumpDir != "" {
+		artifactDir := fmt.Sprintf("ARTIFACT_DIR=%s/gpu-must-gather", dumpDir)
+		mustGatherScriptPath := os.Getenv("PATH_TO_MUST_GATHER_SCRIPT")
+		if mustGatherScriptPath == "" {
+			glog.Error("PATH_TO_MUST_GATHER_SCRIPT environment variable is not set")
+			return
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+		defer cancel()
+		cmd := exec.CommandContext(ctx, mustGatherScriptPath)
+		cmd.Env = append(os.Environ(), artifactDir)
+		output, err := cmd.CombinedOutput()
+		if errors.Is(ctx.Err(), context.DeadlineExceeded) {
+			glog.Errorf("gpu-operator-must-gather.sh script timed out: %v", ctx.Err())
+			return
+		}
+		if err != nil {
+			glog.Errorf("Error running gpu-operator-must-gather.sh script: %v\nOutput: %s", err, output)
```

Comment:
If we extract this into a function, these must be either a generic message or a variable, e.g. `glot.Errorf("%s script timed out: %v", mustGatherScriptPath, ctx.Err()`